include/RAJA/pattern/WorkGroup.hpp

```diff
@@ -375,12 +388,26 @@ struct WorkSite<WorkGroupPolicy<EXEC_POLICY_T,
   using workpool_type = WorkPool<policy, index_type, xarg_type, Allocator>;
   using workgroup_type = WorkGroup<policy, index_type, xarg_type, Allocator>;
 
+private:
+  using workrunner_type = typename workgroup_type::workrunner_type;
+  using per_run_storage = typename workrunner_type::per_run_storage;
+
+  friend workpool_type;
+  friend workgroup_type;
+
+public:
+  using resource_type = typename workpool_type::resource_type;
+
   WorkSite(WorkSite const&) = delete;
   WorkSite& operator=(WorkSite const&) = delete;
 
   WorkSite(WorkSite&&) = default;
   WorkSite& operator=(WorkSite&&) = default;
 
+  RAJA::resources::EventProxy<resource_type> get_event() {
+    return RAJA::resources::EventProxy<resource_type>(&m_resource);
+  }
+
```

Comment:
What do you think about passing the event to WorkSite to store when the WorkGroup returns from run. It could be possible that the event could be called out of order if a user doesn't happen to call get_event explicitly after a run.

Reply:
I agree that's a problem with how this is coded. I was trying to avoid creating an actual event because it may not be used.

Reply:
Would removing get_event from WorkSite and passing an optional Event argument as a ref to the WorkGroup run be a possibility. That could allow the user to explicitly ask for an event. I'm not sure how else you could make it optional for the user to enqueue an event since run returns the WorkSite and not an EventProxy like forall or kernel.

Reply:
I don't think Event is supposed to be default constructable (even if it is). @trws do you have any suggestions of how to cleanly get an event in this case where I'm already returning something else?

Reply:
Let's take another look at this once #1054 is merged.